src/openrct2/ride/TrackData.cpp

```diff
@@ -708,6 +708,349 @@ namespace OpenRCT2::TrackMetaData
 		/* DiagonalBlockBrakes                   */ { 0 },
 	};
 	static_assert(std::size(TrackSequenceProperties) == TrackElemType::Count);
+
+	using WoodenSupportsBySequence = std::array<SequenceWoodenSupport, kMaxSequencesPerPiece>;
+	static constexpr std::array<WoodenSupportsBySequence, TrackElemType::Count> kTrackSequenceWoodenSupports = {
```

Comment:
This is declaring the type twice. For readability, I suggest leaving the declaration type, then removing the `WoodenSupportsBySequence` type from the 339 successive lines, e.g.:

```diff
-	static constexpr std::array<WoodenSupportsBySequence, TrackElemType::Count> kTrackSequenceWoodenSupports = { {
-		WoodenSupportsBySequence{ SequenceWoodenSupport{ WoodenSupportSubType::NeSw } }, // TrackElemType::Flat
-		WoodenSupportsBySequence{ SequenceWoodenSupport{ WoodenSupportSubType::NeSw } }, // TrackElemType::EndStation
-		WoodenSupportsBySequence{ SequenceWoodenSupport{ WoodenSupportSubType::NeSw } }, // TrackElemType::BeginStation
+	static constexpr std::array<WoodenSupportsBySequence, TrackElemType::Count> kTrackSequenceWoodenSupports = {
+		{ SequenceWoodenSupport{ WoodenSupportSubType::NeSw } }, // TrackElemType::Flat
+		{ SequenceWoodenSupport{ WoodenSupportSubType::NeSw } }, // TrackElemType::EndStation
+		{ SequenceWoodenSupport{ WoodenSupportSubType::NeSw } }, // TrackElemType::BeginStation

Note an extra pair of outer braces is required, otherwise it will fail to compile.

I also noticed this file appears to be using tabs for indentation. Could you look into enabling clang-format for this section at least?

Reply:
I’d rather not enable clang-format here, it tends to massively fuck up tables like these.

I’ll look into removing the type. The reason it’s there in the first place is that std::array gets drunk as soon as you do multidimensional stuff - stuff that regular arrays actually get right :')

Reply:
The type inference for std::arrays like these has been improved massively in C++20, so definitely worth giving it another go! :)

Reply:
My struggles are all with C++20.